integration-cli/docker_utils.go

```diff
@@ -463,6 +464,20 @@ func init() {
 			protectedImages[imgTag] = struct{}{}
 		}
 	}
+
+	// Obtain the daemon platform so that it can be used by tests to make
+	// intelligent decisions about how to configure themselves, and validate
+	// that the target platform is valid.
+	res, b, err := sockRequestRaw("GET", "/version", nil, "application/json")
+	defer b.Close()
+	if err != nil || res.StatusCode != http.StatusOK {
+		panic("Init failed to get version " + err.Error() + " " + string(res.StatusCode))
+	}
+	svrHeader, _ := httputils.ParseServerHeader(res.Header.Get("Server"))
+	daemonPlatform = svrHeader.OS
+	if daemonPlatform != "linux" && daemonPlatform != "windows" {
```

Comment:
I'm not sure I see the use of this condition: what matters is that we retrieve the server platform to be able to take decisions accordingly (e.g., skipping a test against a Windows daemon), but I don't see value in being restrictive here.